pkg/basicstation/cups/server.go

```diff
@@ -106,18 +107,34 @@ func (s *Server) getAuth(ctx context.Context, eui types.EUI64, auth string) grpc
 	return s.component.WithClusterAuth()
 }
 
-func (s *Server) getRegistry(ctx context.Context, ids *ttnpb.GatewayIdentifiers) ttnpb.GatewayRegistryClient {
+var errERPeerUnavailable = errors.DefineUnavailable("entity_registry_unavailable", "Entity Registry unavailable for gateway `{gateway_uid}`")
+
+func (s *Server) getRegistry(ctx context.Context, ids *ttnpb.GatewayIdentifiers) (ttnpb.GatewayRegistryClient, error) {
 	if s.registry != nil {
-		return s.registry
+		return s.registry, nil
+	}
+	if peer := s.component.GetPeer(ctx, ttnpb.ClusterRole_ENTITY_REGISTRY, ids); peer != nil {
+		return ttnpb.NewGatewayRegistryClient(peer.Conn()), nil
+	}
+	if ids != nil {
+		return nil, errERPeerUnavailable.WithAttributes("gateway_uid", unique.ID(ctx, ids))
```

Comment:
In fact, `unique.ID()` may still panic if `ids != nil`; it needs the primary identifiers (`GatewayID`); if only `ids.EUI` is set, there's still a panic.

Maybe just remove the attribute